Sources/HStack.swift

```diff
@@ -82,6 +106,109 @@ open class HStack: Stack {
     
     // MARK: helpers
     
+    private func getWidth(
+        for stackable: Stackable,
+        width: CGFloat,
+        currentX: Double,
+        frames: [CGRect]
+    ) -> CGFloat {
+        switch distribution {
+        case .fill:
+            getWidthForDistributionFill(
+                for: stackable,
+                width: width,
+                currentX: currentX,
+                frames: frames
+            )
+        case .fillEqually:
+            getWidthForDistributionFillEqually(
+                for: stackable,
+                width: width
+            )
+        }
+    }
+
+    private func getWidthForDistributionFill(
+        for stackable: Stackable,
+        width: CGFloat,
+        currentX: Double,
+        frames: [CGRect]
+    ) -> CGFloat {
+        let existingItemXOffsetInCommonStack = if let firstFrame = frames.first {
+            firstFrame.origin.x
+        } else {
+           CGFloat(0)
+        }
+
+        // If we have no frames we want to treat the current
+        // x offset as being 0 so our calculations work when
+        // an item is nested in a stack and when it is not
+        //
+        // currentX is the calculation passed in that contains
+        // the width + any spacing that been configured
+        let currentXOffset = if frames.isEmpty {
+            CGFloat(0)
+        } else {
+            CGFloat(currentX)
+        }
+
+        // Calculate the relative X of the item positioned within the given width
+        let relativeXOffsetInStack = currentXOffset - existingItemXOffsetInCommonStack
+
+        if let fixedSizeStackable = stackable as? FixedSizeStackable {
+            let itemWidth = min(
+                fixedSizeStackable.size.width,
+                max(width - relativeXOffsetInStack, 0)
+            )
+
+            #if DEBUG
+                if itemWidth == 0 {
+                    print("\(fixedSizeStackable) - is dropped from layout as it doesn't fit!")
+                }
+            #endif
+
+            return itemWidth
+        } else if let stack = stackable as? Stack {
+            if let fixedSizeStackWidth = stack.width {
+                return fixedSizeStackWidth
+            } else {
+                return width - currentX
+            }
+        }
+        else if let item = stackable as? StackableItem {
+            let itemWidth = min(
+                item.intrinsicContentSize.width,
+                max(width - relativeXOffsetInStack, 0)
+            )
+
+            #if DEBUG
+                if itemWidth == 0 {
+                    print("\(item) - is dropped from layout as it doesn't fit!")
```

Comment:
This will be printed on the console on debug to tell peeps that view being dropped out 🙇‍♂️

Reply:
@ronnychan-seek Would be good to get your thoughts on this.  We deliberated on an assert etc but just like with the UIKit stackview it just fails silently.  We figured a print to console only in DEBUG builds is useful for a dev and wont pollute release builds.  Happy to remove if you see no value.